testsuite/features/step_definitions/command_steps.rb

```diff
@@ -5,6 +5,25 @@
 require 'timeout'
 require 'nokogiri'
 
+When(/^I delete these channels with spacewalk\-remove\-channel:$/) do |table|
+  channels_cmd = "spacewalk-remove-channel "
+  table.raw.each { |x| channels_cmd = channels_cmd + " -c " + x[0] }
+  puts channels_cmd
+  # out, _code = $server.run("spacewalk-remove-channel -l")
+  @command_output, return_code = $server.run(channels_cmd, false)
+  puts return_code
+  puts @command_output
+end
+
+When(/^I list channels with spacewalk\-remove\-channel$/) do
+  @command_output, return_code = $server.run("spacewalk-remove-channel -l")
+  raise "Unable to run spacewalk-remove-channel -l command on server" unless return_code.zero?
+end
+
+Then(/^I should not see "(.*?)" in the output$/) do |arg1|
+  assert(not @command_output.include? arg1)
```

Comment:
This step looks duplicated by that one:
``` 
Then(/^I shouldn't get "([^"]*)"$/) do |arg1|
  found = false
  $command_output.each_line do |line|
    if line.include?(arg1)
      found = true
      break
    end
  end
  raise "'#{arg1}' found in output '#{$command_output}'" if found
end
``` 
Therefore, I don't see the reason to check it line by line, so it could be just fine to refactor it, as you did

Reply:
In fact, now that I notice... we are using in the old step a global variable `$command_output` which can have conflicts when we run features in parallel. We will address it here https://github.com/SUSE/spacewalk/issues/9117
But please, use the old one to don't have duplicated methods.

Reply:
The variable used and the output type in both methods are different. Even the positive method also exists and the logic is different (line 351). I think is safer to let the methods as is now.

Reply:
What do you mean with output type? Steps are not designed to have an output return.
As said in another comment I know the positive method exist, but in fact is so duplicated and we should delete it. By default, we prefer to raise an error if something doesn't match than use asserts.

Reply:
My mistake, I was affecting the wrong variable when I test it.
I remove the method I had and rebuild the existing ones with the implementation that I do, as you suggested.
I will push the commit in a couple of minutes.
Thanks for the review :)